MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelTest.cs

```diff
@@ -1,1637 +1,2306 @@
-                    };
-                    mesh1d.NumNodes = 7;
-
-                    mesh1d.EdgeNodes = new[] { 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6 };
-                    mesh1d.NumEdges = 6;
-                    Assert.IsTrue(api.Mesh1dSet(id, mesh1d));
-
-                    var geometryListIn = new DisposableGeometryList();
-                    geometryListIn.GeometrySeparator = api.GetSeparator();
-                    geometryListIn.XCoordinates = new[] { 5.0, 25.0, 25.0, 5.0 };
-                    geometryListIn.YCoordinates = new[] { 5.0, 5.0, 25.0, 25.0 };
-                    geometryListIn.Values = new[] { 0.0, 0.0, 0.0, 0.0 };
-                    geometryListIn.NumberOfCoordinates = geometryListIn.XCoordinates.Length;
-
-                    var onedNodeMaskPinnedAddress = onedNodeMaskPinned.AddrOfPinnedObject();
-                    Assert.IsTrue(api.ContactsComputeWithPoints(id, ref onedNodeMaskPinnedAddress, ref geometryListIn));
-                    var contacts = api.ContactsGetData(id);
-
-                    // Four contacts are generated, one for each point
-                    Assert.AreEqual(contacts.NumContacts, 4);
-                }
-                finally
-                {
-                    onedNodeMaskPinned.Free();
-                    api.DeallocateState(id);
-                }
-            }
-        }
-
-        [Test]
-        public void GetSelectedVerticesInPolygonThroughAPI()
-        {
-            using (var mesh = GenerateRegularGrid(4, 4, 1, 1))
-            using (var api = new MeshKernelApi())
-            {
-                var id = 0;
-                try
-                {
-                    id = api.AllocateState(0);
-
-                    Assert.IsTrue(api.Mesh2dSet(id, mesh));
-
-                    var geometryListIn = new DisposableGeometryList();
-
-                    geometryListIn.XCoordinates = new[] { 1.5, 1.5, 3.5, 3.5, 1.5 };
-                    geometryListIn.YCoordinates = new[] { -1.5, 1.5, 1.5, -1.5, -1.5 };
-                    geometryListIn.Values = new[] { 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 };
-                    geometryListIn.NumberOfCoordinates = geometryListIn.XCoordinates.Length;
-                    geometryListIn.GeometrySeparator = api.GetSeparator();
-
-                    var selectedVertices = api.GetSelectedVerticesInPolygon(id, ref geometryListIn, 1);
-                    Assert.AreEqual(selectedVertices.Length, 4);
-                }
-                finally
-                {
-                    api.DeallocateState(id);
-                }
-            }
-        }
-    }
-}
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using MeshKernelNETCore.Api;
+using NUnit.Framework;
+using static MeshKernelNETCoreTest.Api.TestUtilityFunctions;
+
+namespace MeshKernelNETCoreTest.Api
+{
+    [TestFixture]
+    [Category("MeshKernelNETTests")]
+    public class MeshKernelTest
+    {
+        [Test]
+        [Category("PerformanceTests")]

```

Comment:
these tests are useless like this. The nature of .NET (JIT) will result in wildly varying results. I created a followup issue to do this in a better way: GRIDEDIT-630.